### DISCUSSION TITLE
how can i get object as data from checkbox array ?

### DISCUSSION THREAD

**Discussion by Eric-Juquel** [react-hook-form/react-hook-form#1737] [General]
Hi , checkbox Array works well for me , but as i map an array of objects to generate my checkbox array , i would like to get array of complete objects with all properties as value instead of an array of string with each object name , i tried many ways without success . do you know how can i manage it please ?

there is the part of my script : 

  `
   { watchTemplateName && getFeaturesByCategory().map((features, i) => {
              return (
                <div className={ `${classes.column} item` } key={ `${features}_${i}` }>
                  <div className={ classes.title }>{ features[0].category }</div>
                  <div className={ classes.modes }>
                    { features.map((feature, i) => {
                      return (
                        <div key={ `${feature}_${i}` }>
                          <label>{ feature.name }</label>
                          <input
                            type='checkbox'
                            name='features'
                            value={ feature.name }
                            ref={ register } />
                        </div>
                      )
                    }) }
                  </div>
                </div>
              )
            }) }`

**Comment by bluebill1049** [comment]
please provide a simple codesandbox for your problem.
